src/test/java/seedu/address/model/tag/TagTest.java

```diff
@@ -26,13 +26,15 @@ public void isValidTagName() {
 
         // Invalid tag names
         assertFalse(Tag.isValidTagName("")); // Empty tag name
-        assertFalse(Tag.isValidTagName("Buyers")); // Tag names should be lowercase
-        assertFalse(Tag.isValidTagName("Sellers")); // Extra space at the end
+        assertFalse(Tag.isValidTagName("Buyers")); // Should not be plural
+        assertFalse(Tag.isValidTagName("Sellers ")); // Extra space at the end
         assertFalse(Tag.isValidTagName("Invalid Tag")); // Space within the tag name
 
         // Valid tag names
         assertTrue(Tag.isValidTagName("buyer"));
         assertTrue(Tag.isValidTagName("seller"));
+        assertTrue(Tag.isValidTagName("buYeR")); // Tag converts string to uppercase
+        assertTrue(Tag.isValidTagName("seLLer")); // Tag converts string to uppercase
```

Comment:
Good comments to let other developers know what the test is about.

Reply:
Good attention to detail!

Reply:
Yes, thank you for adding these comments. Makes the test cases very readable